### ISSUE TITLE
Same dataset, 100% for "single object" but 0% for "multi object"

### ISSUE THREAD

**Issue by MohamadJaber1** [microsoft/singleshotpose#109]
Hello everyone, 

Thank you @btekin for sharing with us this amazing work. I have succeeded in implementing your paper with **single object** for my robotic manipulation task. However, I have difficulties with multiple object pose estimation.

**I would like to tell you what I am suspecting in and please let me know if there is any other changes for the multi object besides the ones I will mention below.**

- **The ground truth bounding box is different than that of single object. It will be shown below, please let me know if thats true and how it should be fixed?**
- **In the main cfg file (as of [occlusion.data](https://github.com/microsoft/singleshotpose/blob/master/multi_obj_pose_estimation/cfg/occlusion.data)), I keeping the directories of one object with its labels and then relying on the  _get_add_objs(--)_ function for calling the other object and training on its images as well. Is it right?**

"More clarification":
For simplicity, I have generated a small dataset of two objects at the same scene with one pose only and for each object dataset, I annontated its respective object in the scene. After training for several epochs, I am showing the output from the validation set where its converge well.

This model is trained for detecting and estimating the pose of a **cat**.
![1908_Cat](https://user-images.githubusercontent.com/48984186/63248748-5336d300-c268-11e9-871a-8b494e4dc366.png)

And this one is trained for detecting and estimating the pose of a **cam**.
![1908_Cam](https://user-images.githubusercontent.com/48984186/63249233-59797f00-c269-11e9-8f2d-de302a6d624a.png)

Now, I am using Python 3.6.8, and PyTorch 1.0.1. 
I have update the following lines in these files according to #30 and [btekin/singleshotpose](https://github.com/btekin/singleshot6Dpose/)

- train_multi.py (configs)
- region_multi.py 
- darknet_multi.py
- dataset_multi.py
- image_multi.py 
changed:
https://github.com/microsoft/singleshotpose/blob/766d5eced25df6f0c438c6414c07b8b686c0558e/multi_obj_pose_estimation/image_multi.py#L24-L27 ...
to:
    if objname == 'cat':
        add_objs = ['cam']
    elif objname == 'cam':
        add_objs = ['cat']
  
Regarding _cfg_ files, I have main one same as [occlusion.data](https://github.com/microsoft/singleshotpose/blob/master/multi_obj_pose_estimation/cfg/occlusion.data) but for two objects. And two individual _cfg_ as [ape.data](https://github.com/microsoft/singleshotpose/blob/master/multi_obj_pose_estimation/cfg/ape_occlusion.data).
The result of the model is:
**Testing "cat"**
![Wrong_Cat](https://user-images.githubusercontent.com/48984186/63251463-153cad80-c26e-11e9-8374-64fdff509103.png)
**Testing "cam"**
![Wrong_Cam](https://user-images.githubusercontent.com/48984186/63251479-18379e00-c26e-11e9-9755-c663d7b80226.png)

**Comment by lmhlll** [comment]
hello, How do I display image results?

**Comment by MohamadJaber1** [comment]
> hello, How do I display image results?
> "Python valid. Py" does not show the resulting image.

https://github.com/microsoft/singleshotpose/blob/766d5eced25df6f0c438c6414c07b8b686c0558e/valid.ipynb

**Comment by lmhlll** [comment]
> > hello, How do I display image results?
> > "Python valid. Py" does not show the resulting image.
> 
> https://github.com/microsoft/singleshotpose/blob/766d5eced25df6f0c438c6414c07b8b686c0558e/valid.ipynb

Thank you! 2019/08/28

**Comment by MrLuer** [comment]
hi， I want my robot grab object，but I dont know how to get the pisition of object relative to camer，for detail see https://github.com/microsoft/singleshotpose/issues/118  
if you can reply I will very appreciate

**Comment by MohamadJaber1** [comment]
> hi， I want my robot grab object，but I dont know how to get the position of object relative to camera，for detail see #118
> if you can reply I will very appreciate @MohamadJaber1

Of course you can. This project (singleshotpose) by @btekin will allow you to get the 6 DoF object pose (position + orientation) with respect to the camera frame. After that you will apply some transformation w.r.t. world coordinate system (robotic environment). I advise to have some literature review before starting and good luck.

**Comment by xiaoxiongmaoxuexi** [comment]
@MohamadJaber1  Thank you for sharing. I have encountered the same problem as you. Have you solved this problem. Thank you for your answer.


Same dataset, 100% for "single object" but 0% for "multi object"
